pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -163,7 +163,10 @@ func (ds *dockerService) RunPodSandbox(ctx context.Context, r *runtimeapi.RunPod
 	cID := kubecontainer.BuildContainerID(runtimeName, createResp.ID)
 	err = ds.network.SetUpPod(config.GetMetadata().Namespace, config.GetMetadata().Name, cID, config.Annotations)
 	if err != nil {
-		// TODO(random-liu): Do we need to teardown network here?
+		// Ensure network resources are cleaned up even if the plugin
+		// succeeded but an error happened between that success and here.
+		ds.network.TearDownPod(config.GetMetadata().Namespace, config.GetMetadata().Name, cID)
```

Comment:
I am not sure if it's OK to skip error of TearDownPod here, maybe a warning at least?

Reply:
@resouer done